scripts/create_test_subset.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python3
+# pylint: disable=too-many-instance-attributes,too-many-locals,unused-argument,no-self-use,wrong-import-order,unused-argument,too-many-arguments
+
+import os
+import subprocess
+from typing import Dict, List
+import random
+import logging
+import click
+from google.cloud import storage
+from sample_metadata import (
+    AnalysisApi,
+    SequenceApi,
+    SampleApi,
+    NewSequence,
+    NewSample,
+    AnalysisModel,
+    exceptions,
+)
+
+logger = logging.getLogger(__file__)
+logging.basicConfig(format='%(levelname)s (%(name)s %(lineno)s): %(message)s')
+logger.setLevel(logging.INFO)
+
+
+sapi = SampleApi()
+aapi = AnalysisApi()
+seqapi = SequenceApi()
+
+
+@click.command()
+@click.option(
+    '--project',
+    required=True,
+    help='The sample-metadata project ($DATASET)',
+)
+@click.option(
+    '-n',
+    required=True,
+    type=int,
+    help='Number of samples to subset',
+)
+def main(
+    project: str,
+    n: int,
+):
+    """
+    Script creates a test subset for a given project.
+    A new project with a prefix -test is created, and for any files in sample/meta,
+    sequence/meta, or analysis/output a copy in the -test namespace is created.
+    """
+    samples = sapi.get_samples(
+        body_get_samples_by_criteria_api_v1_sample_post={
+            'project_ids': [project],
+            'active': True,
+        }
+    )
+    logger.info(f'Found {len(samples)} samples')
+    random.seed(42)  # for reproducibility
+    samples = random.sample(samples, n)
+    sample_ids = [s['id'] for s in samples]
+    logger.info(f'Subset to {len(samples)} samples: {", ".join(sample_ids)}')
+
+    seq_infos: List[Dict] = seqapi.get_sequences_by_sample_ids(sample_ids)
+    seq_info_by_s_id = dict(zip(sample_ids, seq_infos))
+
+    target_project = project + '-test'
+
+    for s in samples:
+        logger.info(f'Processing sample {s["id"]}')
+
+        # processing sample entries
+        try:
+            new_s = sapi.get_sample_by_external_id(s['external_id'], target_project)
+        except exceptions.ApiException:
+            new_s_id = sapi.create_new_sample(
+                project=target_project,
+                new_sample=NewSample(
+                    external_id=s['external_id'] + '-test',
+                    type=s['type'],
+                    meta=_copy_files_in_dict(s['meta'], project),
+                ),
+            )
+        else:
+            new_s_id = new_s.id
+
+        # processing sequence entries
+        seq_info = seq_info_by_s_id.get(s['id'])
+        if seq_info:
+            new_meta = _copy_files_in_dict(seq_info.get('meta'), project)
+            print(new_meta)
+            seqapi.create_new_sequence(
```

Comment:
There's no try-catch here, I don't think you'd need to update if it already exists, but don't want to halt the script.